### DISCUSSION TITLE
Add some kind of scaling for differential thrust on planes with multiple motors

### DISCUSSION THREAD

**Discussion by Painless360** [iNavFlight/inav#9840] [Ideas]
## Current Behaviour
At he moment, the motors can be setup in the mixer and connected separately with a little rudder conrtol mixed into those two channels to provide a little differential thrust. However, there can be issues. At very low throttles, i.e. when coming into land and gliding, a little rudder input can spin up one motor enough and cause a yaw that stalls the aircraft into the ground.

## Desired Behaviour
It would be useful to have some kind of scaling for differential throttle so that when the throttle is below a certain threshold, the effect is reduced and ultimately turned off to avoid these issues. Or some kind of scaling related to throttle to help minimise the potential in flight for an issue.

## Suggested Solution
Ideally a setting in INAV to do this by default would be useful and adding the ability to turn on and off differential thrust via the programming framework would be nice too.

## Who does this impact? Who is this for?
All pilots with models with more than one motor who want to use differential thrust safely.

## Additional context
Thanks for considering this. Issues like the one covered above are why I (and my flying buddies) stopped using the feature in INAV.

**Comment by Helipro** [comment]
That would be nice an on/off capable differential thrust via programing. This is necessary for water planes like my Avios Albatross and other float planes that I fpv fly.  The Albatross on the water requires a lot of differential thrust, with the rudders to keep it straight until the pontoons come out the water.  In the air I prefer to use the rudder mostly with some slight differential thrust.
I propose 3 check boxes next to the motor mixer.  1 box for each profile that can turn on/off differential thrust in that profile.   By switching profiles in flight you can turn differential thrust on and off easily.   

My work around involves edgetx to vary the amount of differential thrust for each mode.  I put the rudder on a separate channel not stabilized by inav so my rudder and differential thrust are not linked and I can turn the differential thrust on and off.

**Comment by MrD-RC** [comment]
The checkboxes are a bit too restrictive. Having an active condition (as the servo mixer does) would allow much more flexibility with the motor mixes. Not just a change for differential thrust. 

With this, you could simply switch between motor mixes, and levels of differential thrust, via an active logic condition. This could be a switch, or even altitude triggered.

**Comment by Helipro** [comment]
> The checkboxes are a bit too restrictive. Having an active condition (as the servo mixer does) would allow much more flexibility with the motor mixes. Not just a change for differential thrust.
> 
> With this, you could simply switch between motor mixes, and levels of differential thrust, via an active logic condition. This could be a switch, or even altitude triggered.

So true!  That would definitely be the best solution. I already use logic for the vtx power and override arming safety checks when craft flying.  I just love inav and the community  enjoying its growth so much over the years!  Thank you guys!

**Comment by ChrisBalmforth** [comment]
Like MrD-RC I want to apply logic conditions to the motor mixer as you can on the servo mixer. My twin-engined fixed-wing plane has undercarriage but no steerable tail wheel so I would like to enable differential thrust on the ground for taxiing and disable it in the air for flying.

**Reply by sensei-hacker** [reply]
This is one of several specific cases of the more general "if the motor mixer had the flexibility of the servo mixer ..."

A stretch goal would be to end up with just _the mixer_.  To have the same flexibility across all outputs. Because there's just one, flexible mixer.

The difficulty here is that the flexible UI used for the servo mixer would get unwieldy for quads, with four motors each having a mix of throttle, roll, pitch, and yaw. That's 16 inputs for four motors.

For planes, we CAN already have the full flexibility of the servo mixer on our motors, achieving what Lee asks for. We can do that by ignoring the motor mixer and setting up our motors on the "servo" mixer.

**Comment by Helipro** [comment]
Now that Inav 7.0 has the ability to change mixer profile on the fly I have:
1st mixer set, for flying, with no differential thrust, and only just rudder.
2nd mixer set with rudder and differential thrust for stunts, taking off on water, and putting around on the water. (manual mode & shared PID's).  My albatross drags a wing pontoon and requires a good amount of differential thrust to drive straight and take off in a canal even with the water rudder assisting.
Now, with mixer profile change, I have a better solution for turning on/off differential thrust than I was using before.

**Reply by ChrisBalmforth** [reply]
That's great! Many thanks for this post, I'll try the same to enable differential thrust for taxiing on the ground on my Twinstar.
